crates/bevy_animation/src/lib.rs

```diff
@@ -593,6 +614,17 @@ fn get_keyframe(target_count: usize, keyframes: &[f32], key_index: usize) -> &[f
     &keyframes[start..end]
 }
 
+// Helper macro for cubic spline interpolation
+// it needs to work on `f32`, `Vec3` and `Quat`
+macro_rules! cubic_spline_interpolation {
+    ($value_start: expr, $tangent_out_start: expr, $tangent_in_end: expr, $value_end: expr, $lerp: expr, $step_duration: expr,) => {
```

Comment:
This is really hard to read, is there a reason why this is a macro instead of a generic function?

Reply:
I'm not sure the type trait constraints can be written. You need V to be `Mul<f32>`, and `<V as Mul<f32>>::Output` to be `
Mul<f32>` and `<V as Mul<f32>>::Output` to be `Add<<V as Mul<f32>>::Output>` and that doesn't seem to work

Reply:
OK sounds like a good justification to me. Could you leave that in a comment? so that we can come back to this later. It doesn't need to be in the doc comment.